app/src/main/java/com/onegini/mobile/exampleapp/network/fcm/FCMRegistrationService.java

```diff
@@ -41,15 +42,31 @@ public FCMRegistrationService(final Context context) {
     storage = new FCMStorage(context);
   }
 
-  public void enrollForPush(final OneginiMobileAuthWithPushEnrollmentHandler enrollmentHandler) {
+  public void enrollForPush(final PushEnrollmentHandler enrollmentHandler) {
     FirebaseApp.initializeApp(context);
-    final String fcmRefreshToken = FirebaseInstanceId.getInstance().getToken();
-    if (fcmRefreshToken == null) {
-      Toast.makeText(context, context.getString(R.string.push_token_is_null_error_message), Toast.LENGTH_LONG).show();
-    } else {
-      final UserClient userClient = OneginiSDK.getOneginiClient(context).getUserClient();
-      userClient.enrollUserForMobileAuthWithPush(fcmRefreshToken, enrollmentHandler);
-    }
+    FirebaseInstanceId.getInstance().getInstanceId().addOnSuccessListener(instanceIdResult -> {
+      final String fcmRefreshToken = instanceIdResult.getToken();
+      if (fcmRefreshToken.isEmpty()) {
+        Toast.makeText(context, context.getString(R.string.push_token_is_null_error_message), Toast.LENGTH_LONG).show();
```

Comment:
In such case nothing is called on the handler. I'd call onError rather than creating Toast (as was suggested in my previous snippet)

Reply:
I would leave both, it's a common mistake to forget about changing the Firebase configuration file.